src/inline-assembly.md

```diff
@@ -413,6 +413,7 @@ The list of clobbered registers for each ABI is updated in rustc as architecture
 Flags are used to further influence the behavior of the inline assembly block.
 Currently the following options are defined:
 - `pure`: The `asm!` block has no side effects, and its outputs depend only on its direct inputs (i.e. the values themselves, not what they point to) or values read from memory (unless the `nomem` options is also set).
+  The `asm!` block must always terminate.
```

Comment:
```suggestion
- `pure`: The `asm!` block has no side effects, must eventually return, and its outputs depend only on its direct inputs (i.e. the values themselves, not what they point to) or values read from memory (unless the `nomem` options is also set).
```
This reads better when placed with the other requirements in the first sentence. Feel free to adjust the exact wording though.